target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4019-meshd01.dts

```diff
@@ -0,0 +1,390 @@
+/* Copyright (c) 2015, The Linux Foundation. All rights reserved.
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+
+/ {
+    model = "Xiaomi Mesh Router D01";
+    compatible = "xiaomi,meshd01", "qcom,ipq4019";
```

Comment:
Based on your name, the compatible would be "xiaomi,mesh-router-d01".
In the forum, it looks to me like the correct name would be "xiaomi,miwifi-mesh-router-d01", though.

Please note that changing this will also require changes to all the cases above, DTS file name, device name in image/Makefile etc.

Reply:
it'a miwifi mesh router (model d01 according to stock firmware) made by xiaomi
i can rename it, but i thought short xiaomi,meshd01 name will be enuff to identify this device

Reply:
Please use the full name, this will be more consistent with the rest and also easier for the user to understand.

Reply:
ive checked dts of other xiaomi boards
miwifi (all xiaomi routers are miwifi) is somwhere skipped and somewhere no
for example xiaomi,mir3g and xiaomi,miwifi-mini
"router" skipped always

so which full name i should use?

Reply:
I'm not sure. BTW, I also wonder what to do with the d01. If it specifies a "submodel" of the mesh router, it should be included in the name. If it's just another naming scheme for the same device, then we shouldn't include it. So, I'd go for (at my current level of understanding)
xiaomi,miwifi-mesh
or
xiaomi,miwifi-mesh-d01

mir3g is older, collapsed names were more frequent back then. BTW, the "r" stands for router there, so technically it's included ;-)